mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,447 @@
+        return;
+    }
+
+    // Make invisible
+    SetModelName(NULL_STRING);
+    SetSolid(SOLID_NONE);
+    m_takedamage = DAMAGE_NO;
+
+    // Pull out a bit
+    if (!CloseEnough(pTrace->fraction, 1.0f))
+    {
+        SetAbsOrigin(pTrace->endpos + (pTrace->plane.normal * 1.0f));
+    }
+
+    Vector vecOrigin = GetAbsOrigin();
+    CBaseEntity *pOwner = GetOwnerEntity();
+
+    float flDamage = GetDamage();
+    float flRadius = GetRadius();
+
+    // Create explosion effect with no damage
+    ExplosionCreate(vecOrigin, GetAbsAngles(), pOwner, flDamage, flRadius, false);
+
+    // Damage
+    CTakeDamageInfo info(this, pOwner, vec3_origin, vecOrigin, flDamage, GetDamageType());
+    RadiusDamage(info, vecOrigin, flRadius, CLASS_NONE, nullptr);
+
+    DestroyTrail();
+
+    m_hOwner = nullptr;
+
+    //if (!pOther->IsPlayer())
+    //{
+    //    UTIL_DecalTrace(pTrace, "Scorch");
+    //}
+
+    UTIL_Remove(this);
+}
+
+void CMomStickybomb::Fizzle(void) { m_bFizzle = true; }
+
+void CMomStickybomb::VPhysicsCollision(int index, gamevcollisionevent_t *pEvent)
+{
+    BaseClass::VPhysicsCollision(index, pEvent);
+
+    int otherIndex = !index;
+    CBaseEntity *pHitEntity = pEvent->pEntities[otherIndex];
+
+    if (!pHitEntity)
+        return;
+
+    bool bIsDynamicProp = (NULL != dynamic_cast<CDynamicProp *>(pHitEntity));
+
+    // Stickybombs stick to the world when they touch it
+    if (pHitEntity && (pHitEntity->IsWorld() || bIsDynamicProp) && gpGlobals->curtime > 0)
+    {
+        m_bTouched = true;
+        //VPhysicsGetObject()->EnableMotion(false);
+
+        // Save impact data for explosions.
+        // m_bUseImpactNormal = true;
+        // pEvent->pInternalData->GetSurfaceNormal(m_vecImpactNormal);
+        // m_vecImpactNormal.Negate();
+    }
+}
+
+void CMomStickybomb::StickybombTouch(CBaseEntity *pOther)
+{
+    Assert(pOther);
+
+    // Don't touch triggers
+    if (pOther->IsSolidFlagSet(FSOLID_TRIGGER | FSOLID_VOLUME_CONTENTS))
+        return;
+
+    // Handle hitting skybox (disappear).
+    const trace_t *pTrace = &GetTouchTrace();
+    if (pTrace->surface.flags & SURF_SKY)
+    {
+        // DestroyTrail();
+
+        m_hOwner = nullptr;
+        UTIL_Remove(this);
+        return;
+    }
+
+    bool bIsDynamicProp = (NULL != dynamic_cast<CDynamicProp *>(pOther));
+
+    // Stickybombs stick to the world when they touch it
+    if (pOther && (pOther->IsWorld() || bIsDynamicProp) && gpGlobals->curtime > 0)
+    {
+        m_bTouched = true;
+        // VPhysicsGetObject()->EnableMotion(false);
+
+        // Save impact data for explosions.
+        // m_bUseImpactNormal = true;
+        // pEvent->pInternalData->GetSurfaceNormal(m_vecImpactNormal);
+        // m_vecImpactNormal.Negate();
+    }
+}
+
+// TODO: Replace with stickybomb trail
+void CMomStickybomb::CreateSmokeTrail()
+{
+    if (m_hRocketTrail)
+        return;
+
+    // Smoke trail.
+    if ((m_hRocketTrail = RocketTrail::CreateRocketTrail()) != NULL)
```

Comment:
You can separate this into assignment and then a null check, much easier to read that way